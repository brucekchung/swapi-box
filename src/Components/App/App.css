.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.load-image {
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.content {
  background-image: url('../../assets/images/background.jpg');
  background-size: cover;
  background-attachment: fixed;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
