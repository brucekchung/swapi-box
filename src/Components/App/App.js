import React, { Component } from 'react'
import { getStarWarsData } from '../../api'
import { cleanData } from '../../cleaner'
import { checkFavorite, addFavorite, removeFavorite } from '../../helper'
import './App.css'

//components
import Banner from '../Banner/Banner'
import Nav from '../Nav/Nav'
import Main from '../Main/Main'

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      favorites: []
    }
  }

  componentDidMount() {
    this.getApiData('films')
  }

  async getApiData(type) {
    const data = await getStarWarsData(type)
    const formatData = await cleanData(data, type)

    this.setState({ [type]: formatData })
  }

  handleClick = (e) => {
    const type = e.target.id
    this.getApiData(type)
  }

  toggleFavorite = (e) => {
    const name = e.target.closest('div').id
    const isFavorite = checkFavorite(this.state, name)

    if (isFavorite) {
      this.setState(removeFavorite(this.state, name))
    } else {
      //also need to set favorite in state.type
      this.setState(addFavorite(this.state, name))
    }
  }

  render() {
    return (
      <section>
        <Banner />
        <Nav handleClick={this.handleClick} />
        <Main allData={this.state} favorite={this.toggleFavorite} />
      </section>
    )
  }
}

export default App