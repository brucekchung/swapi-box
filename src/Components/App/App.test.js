import React from 'react'
import { shallow } from 'enzyme'
import App from './App'

describe('App', () => {
  let wrapper
  const mockFilms = {
    "episode_id": 4,
    "opening_crawl": "It is a period of civil war.\n\nRebel spaceships, striking\n\nfrom a hidden base, have won\n\ntheir first victory against\n\nthe evil Galactic Empire.\n\n\n\nDuring the battle, Rebel\n\nspies managed to steal secret\r\nplans to the Empire's\n\nultimate weapon, the DEATH\n\nSTAR, an armored space\n\nstation with enough power\n\nto destroy an entire planet.\n\n\n\nPursued by the Empire's\n\nsinister agents, Princess\n\nLeia races home aboard her\n\nstarship, custodian of the\n\nstolen plans that can save her\n\npeople and restore\n\nfreedom to the galaxy....",
    "release_date": "1977-05-25",
    "title": "A New Hope",
  }

  beforeEach(async () => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        json: () => Promise.resolve(mockFilms)
      })
    )

    wrapper = await shallow(<App />)
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined()
  })

  // it('should call getApiData on load', () => {
  //   getApiData = jest.fn()
  //   expect(wrapper.instance().getApiData).toHaveBeenCalled()
  // })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

