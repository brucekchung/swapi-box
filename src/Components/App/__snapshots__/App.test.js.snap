// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Banner />,
        <Nav
          handleClick={[Function]}
/>,
        <Main
          allData={
                    Object {
                              "favorites": Array [],
                            }
          }
          favorite={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "allData": Object {
            "favorites": Array [],
          },
          "favorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Banner />,
          <Nav
            handleClick={[Function]}
/>,
          <Main
            allData={
                        Object {
                                    "favorites": Array [],
                                  }
            }
            favorite={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "allData": Object {
              "favorites": Array [],
            },
            "favorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
